diff --git a/lint/stylelint.bzl b/lint/stylelint.bzl
index 0dd34a8..f7b5b7d 100644
--- a/lint/stylelint.bzl
+++ b/lint/stylelint.bzl
@@ -168,8 +168,20 @@ def _stylelint_aspect_impl(target, ctx):
         outputs.human.out,
         outputs.human.exit_code,
         options = color_options,
-        patch = getattr(outputs, "patch", None),
     )
+    if ctx.attr._options[LintOptionsInfo].fix:
+        _exit_code = ctx.actions.declare_file(OUTFILE_FORMAT.format(
+            label = target.label.name, mnemonic = _MNEMONIC, suffix = "patch.exit_code"
+        ))
+        stylelint_action(
+            ctx,
+            ctx.executable,
+            files_to_lint,
+            None,
+            _exit_code,
+            options = color_options,
+            patch = outputs.patch,
+        )
 
     raw_machine_report = ctx.actions.declare_file(OUTFILE_FORMAT.format(label = target.label.name, mnemonic = _MNEMONIC, suffix = "raw_machine_report"))
 
